reqbaz/src/main/java/de/rwth/dbis/acis/bazaar/service/resources/WebhookResource.java

```diff
@@ -0,0 +1,243 @@
+                )
+        ),
+        schemes = SwaggerDefinition.Scheme.HTTPS
+)
+@Path("/webhook/{projectId}/github")
+public class WebhookResource {
+
+    private L2pLogger logger = L2pLogger.getInstance(WebhookResource.class.getName());
+    private BazaarService bazaarService;
+
+    private static final char[] HEX = {
+            '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'
+    };
+
+    public WebhookResource() throws Exception {
+        bazaarService = (BazaarService) Context.getCurrent().getService();
+    }
+
+    /**
+     * This method processes the payload sent from GitHub
+     *
+     * @param projectId  projectId
+     * @param requestBody  payload body
+     * @param eventHeader   github event header
+     * @param signatureHeader github signature header
+     * @return Response
+     */
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Webhook Endpoint")
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpURLConnection.HTTP_CREATED, message = "Returns OK"),
+            @ApiResponse(code = HttpURLConnection.HTTP_NOT_FOUND, message = "Not found"),
+            @ApiResponse(code = HttpURLConnection.HTTP_INTERNAL_ERROR, message = "Internal server problems")
+    })
+    public Response handleWebhook(@PathParam("projectId") int projectId, String requestBody,
+                                  @HeaderParam("X-GitHub-Event")String eventHeader,
+                                  @HeaderParam("X-Hub-Signature-256")String signatureHeader
+    ){
+
+        String issueHtml = "";
+        String issueBody = "";
+        String issueNumber = "";
+        String issueBodyMsgResult = null;
+        String hookId = "";
+        String pullRequestHtml = "";
+        String releaseHtmlUrl = "";
+        String action = "";
+
+        try {
+            // parse the payload
+            JSONParser p = new JSONParser(JSONParser.MODE_PERMISSIVE);
+            JSONObject j = (JSONObject) p.parse(requestBody);
+
+            // DB Operations
+            DALFacade dalFacade = null;
+
+            String registrarErrors = bazaarService.notifyRegistrars(EnumSet.of(BazaarFunction.VALIDATION,BazaarFunction.USER_FIRST_LOGIN_HANDLING));
+            if(registrarErrors != null){
+                ExceptionHandler.getInstance().throwException(ExceptionLocation.BAZAARSERVICE, ErrorCode.UNKNOWN, registrarErrors);
+            }
+
+            Agent agent = Context.getCurrent().getMainAgent();
+            String userId = agent.getIdentifier();
+
+            dalFacade = bazaarService.getDBConnection();
+            Integer internalUserId = dalFacade.getUserIdByLAS2PeerId(userId);
+            Project projectToReturn = dalFacade.getProjectById(projectId, internalUserId);
+
+            // Validate the x-hub-signature-256
+            if(signatureHeader != null) {
```

Comment:
Verifying the signature should be refactored into a separate method so these if blocks don't grow so much and are more readable